next/TilesetWithFullMetadata/TilesetWithFullMetadata.json

```diff
@@ -0,0 +1,1318 @@
+{
+  "extensions" : {
+    "3DTILES_metadata" : {
+      "schema" : {
+        "classes" : {
+          "exampleClass" : {
+            "name" : "Full example metadata class",
+            "description" : "An example metadata class with all property types",
+            "properties" : {
```

Comment:
This example makes me think "exhaustive", not "representative". I don't think we need to list every possible combination, just an assortment of the important cases. for example, just have 1-2 of these categories:

* basic numeric properties
* normalized properties
* vector properties
* matrix property
* strings
* array (fixed length)
* array (variable length)
* enum
* 64-bit types
* a few (emphasis on few) examples that combine the above (e.g. arrays of vectors, normalized vectors, array of string).

Reply:
I considered this, and also created one "representative" example, with properties of these types
```
        addExampleProperty(metadataClass, "SINGLE", "INT8", null, null);
        addExampleProperty(metadataClass, "SINGLE", "FLOAT32", null, null);
        addExampleProperty(metadataClass, "VEC3", "FLOAT32", null, null);
        addExampleProperty(metadataClass, "SINGLE", "STRING", null, null);
        addExampleProperty(metadataClass, "SINGLE", "ENUM", null, null);
        addExampleProperty(metadataClass, "ARRAY", "INT32", null, null);
        addExampleProperty(metadataClass, "ARRAY", "INT32", 5, null);
        addExampleProperty(metadataClass, "ARRAY", "ENUM", null, null);
        addExampleProperty(metadataClass, "ARRAY", "STRING", null, null);
```
That should already cover most use cases. 

I wasn't sure what the "exhaustive" vs. "representative" referred to. Having an idea of all possible types may not be the worst thing, and shed some light on corner cases, like number encodings, or ... maybe also the fact that

> e.g. arrays of vectors,

is currently not possible.